### ISSUE TITLE
Not playing x265 output?

### ISSUE THREAD

**Issue by charlescva** [strukturag/vlc-libde265#7]
Hi, I'm a bit new to HEVC but I'm having an interesting issue and I thought you may be able to help.

Using VLC media player 2.2.4 Weatherwax (revision 2.2.3-37-g888b7e89) on Debian Jessie, Kernel 4.7.0

I compiled your plugin and copied the two .so files to /usr/lib/vlc/plugins path:

> charles@box:/usr/lib/vlc/plugins$ md5sum codec/libde265_plugin.so 
> 3e001fec3976dba4d0ae136a43270370  codec/libde265_plugin.so
> charles@box:/usr/lib/vlc/plugins$ md5sum demux/libde265demux_plugin.so 
> 8ad1b7a0e955713a469b4ca496e021b5  demux/libde265demux_plugin.so

As a test, I downloaded the "Tears of Steel" video from http://www.libde265.org/downloads-videos/ and it plays great!

I am now trying to create my own HEVC encoded video and VLC is not playing it.  Instead it shows a blank screen for about a second and then hits the "EOF".

In a nutshell, I am recording from /dev/video0 with ffmpeg:
> ffmpeg -i /dev/video0 -c:v rawvideo out.yuv

Encoding it with x265:
> x265 out.yuv --input-res 640x480 --fps 30 -o out.mkv

And playing it back with your Plugin on VLC:
> cvlc out.mkv

The VLC Log is quite verbose but I have pasted it here: [http://pastebin.com/JdaXJzz1](http://pastebin.com/JdaXJzz1)

I understand that you do not support x265, but I'm hoping you can give me some guidance as to what I may be doing wrong overall.

Thanks,
Charles

**Comment by charlescva** [comment]
UPDATE: ffplay will attempt to "play" the video.  But it looks really messed up.  It seems to be flickering on a single frame, and sort of gray and scaled wrong, with patterns of RGB dots?

It doesnt seem to play the YUV file generated from ffmpeg, so I'm going to try some other ffmpeg commands.  Perhaps there is an issue pulling 'rawvideo' from the camera.

**Comment by charlescva** [comment]
So after Compiling FFMPEG with the GPL flag and H264 encoder flag enabled to test another method, 

> ffmpeg -i /dev/video0 -c:v libx265 -preset medium -crf 28 -c:a aac -b:a 128k output.mp4

I get the following output from VLC after a successful encoding,

[00007fac2c0064b8] vdpau_display vout display error: source video YCbCr format not supported
Segmentation fault

**Comment by charlescva** [comment]
WooHoo!!! I finally figured it out.  Needed to set it to YUV 4:2:0 Chroma subsampling:

ffmpeg -i /dev/video0 -c:v libx265 -vf format=yuv420p output.mp4
